addons/mail/static/src/utils/common/hooks.js

```diff
@@ -579,3 +580,58 @@ export const useMovable = makeDraggableHook({
         return { top, left };
     },
 });
+
+export function useLongPress(refName, callback) {
+    if (!hasTouch()) {
+        return;
+    }
+    const MOVE_TRESHOLD = 10;
+    const DELAY = 500;
+    const ref = useRef(refName);
+    let timer = null;
+    let startX = 0;
+    let startY = 0;
+    useLazyExternalListener(
+        () => ref.el,
+        "pointerdown",
+        (ev) => {
+            startX = ev.clientX;
+            startY = ev.clientY;
+            timer = setTimeout(() => {
+                callback(ev);
+                timer = null;
+            }, DELAY);
+        }
+    );
+    useLazyExternalListener(
+        () => ref.el,
+        "pointermove",
+        (ev) => {
+            if (!timer) {
+                return;
+            }
+            const dx = ev.clientX - startX;
+            const dy = ev.clientY - startY;
+            if (Math.hypot(dx, dy) > MOVE_TRESHOLD) {
+                clearTimeout(timer);
+            }
+        }
+    );
+    useLazyExternalListener(
+        () => ref.el,
+        "pointerup",
+        () => {
+            clearTimeout(timer);
+            timer = null;
+        }
+    );
+    useLazyExternalListener(
+        () => ref.el,
+        "pointercancel",
+        () => {
+            clearTimeout(timer);
+            timer = null;
+        }
+    );
+    return ref;
```

Comment:
Couldn't you use touch events instead?
There are laptops with touchscreen and trackpad, therefore the scroll may be made with either of them. We wouldn't want long click from trackpad/mouse being considered a touch long press because the machine has a touch screen.

Maybe inter-leave some touch events listeners that are required for the resulting function call of the long press.